testng-core/src/test/kotlin/test/SimpleBaseTest.kt

```diff
@@ -0,0 +1,407 @@
+package test
+
+import org.assertj.core.api.Assertions.assertThat
+import org.testng.*
+import org.testng.annotations.ITestAnnotation
+import org.testng.internal.annotations.AnnotationHelper
+import org.testng.internal.annotations.DefaultAnnotationTransformer
+import org.testng.internal.annotations.IAnnotationFinder
+import org.testng.internal.annotations.JDK15AnnotationFinder
+import org.testng.xml.*
+import org.testng.xml.internal.Parser
+import java.io.*
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+
+open class SimpleBaseTest {
+
+    companion object {
+        private const val TEST_RESOURCES_DIR = "test.resources.dir"
+
+        @JvmStatic
+        fun run(vararg testClasses: Class<*>) = run(false, *testClasses)
+
+        @JvmStatic
+        private fun run(skipConfiguration: Boolean, tng: TestNG) =
+            InvokedMethodNameListener(skipConfiguration).apply {
+                tng.addListener(this)
+                tng.run()
+            }
+
+        @JvmStatic
+        fun run(skipConfiguration: Boolean, vararg testClasses: Class<*>) =
+            run(skipConfiguration, create(*testClasses))
```

Comment:
From a Kotlin POV, I will be more confortable with: `TestNG().run(skipConfiguration, *testClasses)` or `create(*testClasses).run(skipConfiguration)`  because we are supposed to run test against the testng instance.

The Java parent classe doesn't do that because extensions are not possible in Java.

Reply:
@juherr - From what I have read about extension functions in Kotlin, they get translated into `static` funtions in the Java world, with the type (to which they are added as extension funtions) getting passed implicitly as the first parameter to the static method.

The problem with this is that now all the Java sub classes will be required to start adding the extra static parameter which is going to add a whole lot of changes.

So can we for now just stick to the `@JvmStatic` way of exposing the companion methods and then move forward with migrating the tests into Kotlin. I will take care of incremental refactoring of the tests to be more Kotlinish, once the migration is complete, because that is going to be a lot more easier.